### PR TITLE
fix(): resolve `yarn build` with assets hangs

### PR DESCRIPTION
**Pull request by Dominic-Preap**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: #608

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by Dominic-Preap** [comment]
@kamilmysliwiec I fixed the issue by add closing asset watcher on webpack when using `nest build`.

File: lib/compiler/webpack-compiler.ts [hunk 1]
```diff
@@ -80,6 +83,7 @@ export class WebpackCompiler {
         warningsFilter: /^(?!CriticalDependenciesWarning$)/,
       });
       if (!err && !stats.hasErrors()) {
+        !onSuccess && assetsManager.closeWatchers(); // if use `nest build` on webpack, close assets watchers
         onSuccess && onSuccess();
```
Comment (kamilmysliwiec) [inline lib/compiler/webpack-compiler.ts]:
```suggestion
        if (!onSuccess) { 
           assetsManager.closeWatchers(); // if use `nest build` on webpack, close assets watchers
        } else {
           onSuccess();
        }
```

Comment (Dominic-Preap) [reply lib/compiler/webpack-compiler.ts]: already updated.

**Comment by kamilmysliwiec** [comment]
Thank you @Dominic-Preap!

**Pull request merged**